### PR TITLE
[inline] Clone return range attribute on the callsite into inlined call

### PR DESCRIPTION
**Pull request by andjo403**
CC @goldsteinn @nikic

### CODE REVIEW

**Comment by andjo403** [comment]
hmm noticed #91101 now looks like I need to coordinate with that PR

**Comment by andjo403** [comment]
It was the usage of exactIntersectWith in https://github.com/llvm/llvm-project/pull/91101 vs the intersectWith that I used that I wanted to sort out but think that we have conluded that it shall be intersectWith.
so this it ready for review

**Review by nikic — APPROVED** [review]
LGTM

File: llvm/lib/Transforms/Utils/InlineFunction.cpp [hunk 1]
```diff
@@ -1444,6 +1445,8 @@ static AttrBuilder IdentifyValidPoisonGeneratingAttributes(CallBase &CB) {
     Valid.addAttribute(Attribute::NonNull);
   if (CB.hasRetAttr(Attribute::Alignment))
     Valid.addAlignmentAttr(CB.getRetAlign());
+  if (CB.hasRetAttr(Attribute::Range))
+    Valid.addRangeAttr(*CB.getRange());
```
[the remark this answers is not in the document]
Comment (andjo403) [reply llvm/lib/Transforms/Utils/InlineFunction.cpp]: fixed

**Comment by andjo403** [comment]
Comment fixed and need help with merging.

**Comment by aeubanks** [comment]
for future reference, https://github.com/llvm/llvm-project/commit/3cd67eeca28ab1084d02b7976de1af4c4c8d37d5 fixes a miscompile that this patch uncovered

**Pull request merged**